Java/Varargs/Overloading_Vararg_Methods/Java 11.2: Overloading Vararg Methods #2218.md

```diff
@@ -0,0 +1,153 @@
+<h2 align="center"> Overloading Varags Method </h2>
+<hr>
+
+
+Brief explanation of **Method Overloading.**
+
+A method overloading is a technique in which multiple methods with same name but different types and numbers of arguments are used.
+e.g. 
+    
+    int foo(int i);
+    
+    int foo(int i, float j);
+    
+    float foo(float i, float j);
+
+**NOTE:- Only changing the return type of method is not method overloading.**
+
+<br/>
+
+**Method Overloading in Varargs**
+
+Here the method overloading is done in two ways:- 
+  
+  1. Overloading when methods have only varargs
+     
+    public static void main(String[] args)
+	  {
+	  	fun();
+	  }
+
+	  //varargs method with float datatype
+	  static void fun(float... x)
+	  {
+	  	System.out.println("float varargs");
+  	}
+	
+	  //varargs method with int datatype
+	  static void fun(int... x)
+	  {
+	  	System.out.println("int varargs");
+	  }
+	
+	  //varargs method with double datatype
+	  static void fun(double... x)
+	  {
+		  System.out.println("double varargs");
+	  }
+	  
+   ![Screenshot 2021-09-24 at 2 09 13 PM](https://user-images.githubusercontent.com/70788113/134648966-6eb99a3c-32fc-47c3-b396-d8f6814090a3.png)
+	  
+  
+    Here Java will call the function using the rule that the most specific method is chosen according to type promotion.    
+    This means the type of varargs which is more specific according to the rule will be called.    
+    The following rules define the direct supertype relation among the primitive types in this case:
+    
+   <ul>
+    
+   <li>double > float</li>
+    
+   <li>float > long</li>
+    
+   <li>long > int</li>
+    
+   <li>int > char</li>
+    
+   <li>int > short</li>
+    
+   <li>short > byte</li>
+   
+   </ul>
+    
+    
+    So the output of this code snippet will be **_int varargs_**, because int is more specific than double and float.
```

Comment:
Update this part as well, it is appearing as **_int varargs_**.

Reply:
Done! Anything else?

Reply:
Looks fine to me now.
@garvitraj - Please have a final look.
Thanks